### ISSUE TITLE
version 18.12 does not compile with xmpp v1.2.7

### ISSUE THREAD

**Issue by marcelloceschia** [processone/ejabberd#2725]
Try to compile v18.12 using the default rebar.config will result in the following issue:

```
===> Compiling src/XmppAddr.erl failed
src/XmppAddr.erl:44: illegal pattern
src/XmppAddr.erl:45: variable 'Exception' is unbound
src/XmppAddr.erl:47: variable 'Stk' is unbound
src/XmppAddr.erl:49: variable 'Stk' is unbound
src/XmppAddr.erl:58: illegal pattern
src/XmppAddr.erl:59: variable 'Exception' is unbound
src/XmppAddr.erl:61: variable 'Stk' is unbound
src/XmppAddr.erl:63: variable 'Stk' is unbound
```

upgrading xmpp dependency to version 1.2.8 does fix this issue

**Comment by licaon-kter** [comment]
I built it ok for me iirc.

Was this a clean build or an update?

What OS?

**Comment by marcelloceschia** [comment]
It was a clean docker build from alpine:3.8 
OTP: 20.3.4

The build is done by gitlab-ci pipline, I can send you the build log if you need it.

Build steps:
```
VERSION="18.12"
apk upgrade --update musl
apk add build-base git zlib-dev openssl-dev yaml-dev expat-dev sqlite-dev gd-dev jpeg-dev libpng-dev libwebp-dev autoconf automake elixir erlang-crypto erlang-eunit erlang-mnesia erlang-erts erlang-hipe erlang-tools erlang-os-mon erlang-syntax-tools erlang-parsetools erlang-runtime-tools erlang-reltool file curl
mix local.hex --force
mix local.rebar --force
git clone https://github.com/processone/ejabberd.git source
cd source && git checkout tags/${VERSION}
cp ../vars.config . && cp ../rel/*exs rel/
mix deps.get
MIX_ENV=prod mix do compile
```

**Comment by prefiks** [comment]
Hello,

This problem is fixed in xmpp 1.2.8 (hex.pm release of 1.2.7 did contain generated files that were created by erlang R21, and they didn't compile on older erlang versions, new release did remove those and added commands to generate those files just before compilation).

You could update just that dep with "mix deps.update xmpp", or clone from rel-18.12.1 branch that has this dep updated.
